### PR TITLE
chore: cherry-pick 7abc7e45b2 from node

### PR DESCRIPTION
**Pull request by indutny**
#### Description of Change

Backports: https://github.com/nodejs/node/pull/38506
See: https://github.com/electron/electron/pull/28957 for discussion

In few words, `napi_threadsafe_function` was significantly slower in Electron due to scheduling each call on the next uv tick. We've patched electron in #28957 to let microtasks queue execute from within `uv_run()` and now this backports a patch that makes `napi_threadsafe_function` calls scheduled while one was running happen on the same tick. Fixing the performance issue.

cc @codebytere @deepak1556 @zcbenz 

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [ ] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [ ] relevant documentation is changed or added
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Improved performance of `napi_threadsafe_function`

### CODE REVIEW

**Comment by indutny** [comment]
Would appreciate if it could get backported into 12-x-y branch.

**Review by deepak1556 — APPROVED** [review]

**Comment by indutny** [comment]
Wow, thanks for quick LGTM @deepak1556 What's the procedure here? Do we need more LGTMs, or is this PR good to merge already?

**Comment by deepak1556** [comment]
Yup a PR needs atleast 2 LGTM before landing, usually a 24hr period should suffice to cover that. This should be good to merge tomorrow.

**Review by zcbenz — APPROVED** [review]

**Comment by deepak1556** [comment]
Failing test is unrelated, merging.

**Comment by indutny** [comment]
Yay, awesome! Thanks!

**Pull request merged**
